dashboard/src/components/DeploymentForm/DeploymentForm.tsx

```diff
@@ -212,6 +217,32 @@ class DeploymentForm extends React.Component<IDeploymentFormProps, IDeploymentFo
       </div>
     );
   };
+
+  private setBasicForm = (enable: boolean) => {
+    // OnClick requires to return a function
+    return () => {
+      this.setState({ showBasicForm: enable });
+    };
+  };
+
+  private renderTabs = () => {
```

Comment:
Since we're going to use it only in this component, I'm fine with having the tabs defined inside it. However, if we plan to repeat this pattern in other places of the app, we should make it an independent component.